assets/app/scripts/directives/oscObjectDescriber.js

```diff
@@ -0,0 +1,122 @@
+angular.module('openshiftConsole')
+  .directive('oscObjectDescriber', function(ObjectDescriber) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: 'views/directives/osc-object-describer.html',
+      link: function(scope, elem, attrs) {
+        var callback = ObjectDescriber.onResourceChanged(function(resource, kind) {
+          scope.$apply(function() {
+            scope.kind = kind;
+            scope.resource = resource;
+          });
+        });
+        scope.$on('$destroy', function() {
+          ObjectDescriber.removeResourceChangedCallback(callback);
+        });    
+      }
+    };
+  })
+  .directive('oscObject', function(ObjectDescriber) {
+    return {
+      restrict: 'AC',
+      scope: {
+        resource: '=',
+        kind: '@'
+      },
+      link: function(scope, elem, attrs) {
+        $(elem).on("click.oscobject", function() {
+          if (scope.resource) {
+            ObjectDescriber.setObject(scope.resource, scope.kind || scope.resource.kind, {source: scope});
+            // Have to stop event propagation or nested resources will fire parent handlers
+            return false;
+          }
+        });
+
+        $(elem).on("mousemove.oscobject", function() {
+          if (scope.resource) {
+            $(".osc-object-hover").removeClass("osc-object-hover");
```

Comment:
@liggitt 
What about:
```
$(".osc-object-hover").not(this).removeClass("osc-object-hover");
$(this).addClass("osc-object-hover");
```
pretty sure the addClass under the covers only adds the class if its not already there